enterprise/api/filters.py

```diff
@@ -33,25 +36,68 @@ def filter_queryset(self, request, queryset, view):
 
 class EnterpriseCustomerUserFilterBackend(filters.BaseFilterBackend):
     """
-    Allow filtering based on user's email or username on enterprise customer user api endpoint.
+    Allow filtering on the enterprise customer user api endpoint.
     """
 
+    def _filter_by_user_attributes(self, request, queryset):
+        """
+        Filter queryset by email or username.
+        """
+        email = request.query_params.get('email', None)
+        username = request.query_params.get('username', None)
+
+        query_params = {}
+
+        if email:
+            query_params.update(email=email)
+        if username:
+            query_params.update(username=username)
+
+        if query_params:
+            users = User.objects.filter(**query_params).values_list('id', flat=True)
+            return queryset.filter(user_id__in=users)
+
+        return queryset
+
+    def _filter_by_enterprise_attributes(self, request, queryset):
+        """
+        Filter queryset by enterprise_customer_uuid or enterprise role.
+        """
+        enterprise_customer_uuid = request.query_params.get('enterprise_customer_uuid', None)
+        role = request.query_params.get('role', None)
+
+        query_params = {}
+
+        if enterprise_customer_uuid:
+            query_params.update(enterprise_customer_id=enterprise_customer_uuid)
+
+        if role:
+            role_assignment_filters = {'role__name': role}
+
+            if not enterprise_customer_uuid:
+                raise ValidationError('Cannot filter by role without providing enterprise_customer_uuid.')
+
+            if enterprise_customer_uuid:
+                role_assignment_filters.update(enterprise_customer_id=enterprise_customer_uuid)
```

Comment:
Nit: `enterprise_customer_uuid` is guaranteed to be truthy at this point.

Reply:
:facepalm forgot to remove